pkg/rpc/client/wsclient.go

```diff
@@ -266,12 +266,12 @@ func (c *WSClient) getResponseChannel(id uint64) chan *response.Raw {
 
 func (c *WSClient) makeWsRequest(r *request.Raw) (*response.Raw, error) {
 	ch := make(chan *response.Raw)
-	c.registerRespChannel(r.ID, ch)
-
 	select {
 	case <-c.done:
 		return nil, errors.New("connection lost before sending the request")
-	case c.requests <- r:
+	default:
+		c.registerRespChannel(r.ID, ch)
+		c.requests <- r
```

Comment:
So we can lock here and miss `c.done` signal.

Reply:
Hm. See the updated version, it should be fixed there.